app/controllers/workflow_queues_controller.rb

```diff
@@ -22,12 +18,63 @@ def all_queued
     @workflow_steps = @workflow_steps.where(hours_ago)
   end
 
+  # Used by robot-master:
```

Comment:
There's a lot going on here inside the controller. I have three reactions:

1. Do you feel this is the best place for the SQL/ARel code to live? (Could some of this live in the model?)
2. I'm surprised by how much of this you had to roll your own of. (Not criticism, just surprise at having to do manual query parsing and scope logic.)
2. Wherever this lives, could you add more documentation about what all is going on here?

Reply:
1. I extracted it to IntersectQuery
2. Me too.
3. Added